### DISCUSSION TITLE
Prefill an `email_field` with a template expression value

### DISCUSSION THREAD

**Discussion by fluca1978** [mojolicious/mojo#2159] [Q&A] [answered]
How can I prefill an `email_field` with the value that comes out from a Perl expression?
I've tried:

```
%= email_field 'email' => <%= $user && $user->is_logged_in ? $user->email : '' %>,  id => 'email', class => 'form-control'
```

and it partially works, since the email field is generated and filled, but it appears with all the other parts of the field configuration:

```
 <input name="email" type="email" value="foo@bar.com">,  id => 'email', class => 'form-control' %>
```

seems like a wrong escaping due to the expression ending tag.

**Comment by fluca1978** [comment, marked as answer]
I think I've solved by myself, but I would like to get a confirm or suggestion if this is not the best solution.
Since `%=` starts a Perl line evaluation, I do not need to embed `<%=` and I can insert the perl expression in the line:

```
%= email_field 'email' => ( $user && $user->is_logged_in ? $user->email : '' ),  id => 'email', class => 'form-control'
```

which produces:

```
<input class="form-control" id="email" name="email" type="email" value="foo@bar.com">
```

**Reply by ghandmann** [reply]
You are allowed to create local variables. So you could put the result of your expression into a variable and use that variable as value for the `email_field`:

```perl
% my $expression_result = "do something";
%= email_field 'email' => $expression_result;
```

**Reply by fluca1978** [reply]
Thanks @ghandmann, this is useful for more complex expressions!
